control-plane/src/modules/workflows/agent/run.ts

```diff
@@ -284,24 +329,52 @@ async function findRelatedFunctionTools(workflow: Run, search: string) {
 
   const toolContexts = await Promise.all(
     relatedTools.map(async (toolDetails) => {
-      const metadata = await getToolMetadata(
-        workflow.clusterId,
-        toolDetails.serviceName,
-        toolDetails.functionName,
-      );
+      const [metadata, resolvedJobs, rejectedJobs] = await Promise.all([
+        getToolMetadata(
+          workflow.clusterId,
+          toolDetails.serviceName,
+          toolDetails.functionName,
+        ),
+        getLatestJobsResultedByFunctionName({
+          clusterId: workflow.clusterId,
+          service: toolDetails.serviceName,
+          functionName: toolDetails.functionName,
+          limit: 3,
+          resultType: "resolution",
+        }).then((jobs) => {
+          return jobs?.map((j) => ({
+            targetArgs: anonymize(j.targetArgs),
+            result: anonymize(j.result),
+          }));
+        }),
+        getLatestJobsResultedByFunctionName({
+          clusterId: workflow.clusterId,
+          service: toolDetails.serviceName,
+          functionName: toolDetails.functionName,
+          limit: 3,
+          resultType: "rejection",
+        }).then((jobs) => {
+          return jobs?.map((j) => ({
+            targetArgs: anonymize(j.targetArgs),
+            result: anonymize(j.result),
+          }));
+        }),
+      ]);
 
       const contextArr = [];
 
-      if (metadata?.additionalContext) {
-        contextArr.push(`<context>${metadata.additionalContext}</context>`);
+      const successJobsContext = formatJobsContext(resolvedJobs, "success");
```

Comment:
We should probably put all these inside some kind of tag that explains they are _previous_ examples of the function being called?

Reply:
Shouldn't be a problem as the actual values are removed, but the model may see them as previous job calls.

Reply:
This should be clearer now.